code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/internal/AssurancePluginScreenshot.java

```diff
@@ -89,26 +81,94 @@ private void getCurrentScreenShot(final CaptureScreenShotListener captureScreenS
             return;
         }
 
-        // create bitmap screen capture
+        // Get current activity
         final Activity currentActivity =
                 ServiceProvider.getInstance().getAppContextService().getCurrentActivity();
 
-        if (currentActivity != null) {
-            currentActivity.runOnUiThread(
-                    new Runnable() {
-                        @Override
-                        public void run() {
-                            View currentWindow =
-                                    currentActivity.getWindow().getDecorView().getRootView();
-                            currentWindow.setDrawingCacheEnabled(true);
-                            Bitmap bitmap = Bitmap.createBitmap(currentWindow.getDrawingCache());
-                            currentWindow.setDrawingCacheEnabled(false);
-
-                            if (captureScreenShotListener != null) {
-                                captureScreenShotListener.onCaptureScreenshot(bitmap);
-                            }
+        if (currentActivity == null) {
+            Log.error(
+                    Assurance.LOG_TAG,
+                    LOG_TAG,
+                    "Unable to take screenshot, current activity is null.");
+            return;
+        }
+
+        // Check if device supports PixelCopy API (Android 8.0 and above)
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            Log.error(
+                    Assurance.LOG_TAG,
+                    LOG_TAG,
+                    "Screenshot not supported on Android versions below 8.0 (API 26)");
+            if (parentSession != null) {
+                parentSession.logLocalUI(
+                        UILogColorVisibility.LOW,
+                        "Screenshot not supported on Android versions below 8.0 (API 26)");
+                sendErrorEvent("Screenshot not supported on Android versions below 8.0 (API 26)");
+            }
```

Comment:
This will regress existing screenshot compatibility with `API 21` to `API 26` that we currently support (albeit not the hardware bitmaps). 
Can we fallback to default screenshot capturing for `21 <= API <= 25` and then fail with an error event if it turns out to be a hardware bitmap?

```
if (Build.VERSION.SDK_INT >= 21 && Build.VERSION.SDK_INT <= 25 ) {
   try {
     // success for software bitmap
   } catch (e) {
      // fail here for hardware bitmap
     }
 
} else {
      
      // capture screenshot with PixelCopy

}


```